src/widget/tree_structure.rs

```diff
@@ -0,0 +1,191 @@
+use std::collections::HashMap;
+
+use xilem_core::VecSplice;
+
+use crate::{
+    id::Id,
+    widget::{ChangeFlags, Pod},
+};
+
+use crate::view::ElementsSplice;
+
+/// The pure structure (parent/children relations via ids) of the widget tree.
+#[derive(Debug, Default, Clone)]
+pub struct TreeStructure {
+    parent: HashMap<Id, Option<Id>>,
+    // TODO this is taken from the druid lasagna branch, is the Option here intentional, i.e. to track all parentless ids with None?
+    children: HashMap<Option<Id>, Vec<Id>>,
+}
```

Comment:
@raphlinus Since this is copied from your lasagna branch, do you have an explanation for that TODO?

Reply:
I'm pretty sure `None` as a key to the children map means "return the root widget". I think you can skip it and have `HashMap<Id, Vec<Id>>` instead.

Reply:
Yeah there's no `Option` necessary for both of these, since it's only the relations. Updated that in the latest commits.